janus_server/src/datastore.rs

```diff
@@ -830,27 +841,30 @@ impl<C: Clock> Transaction<'_, C> {
     /// aggregation job. It returns an error if the aggregation job has no current lease.
     pub async fn release_aggregation_job(
         &self,
-        task_id: TaskId,
-        aggregation_job_id: AggregationJobId,
+        lease: &Lease<AcquiredAggregationJob>,
     ) -> Result<(), Error> {
-        // TODO(brandon): sanity-check that the aggregation job is leased?
         let stmt = self
             .tx
             .prepare_cached(
-                "UPDATE aggregation_jobs SET lease_expiry = TIMESTAMP '-infinity'
+                "UPDATE aggregation_jobs SET lease_expiry = TIMESTAMP '-infinity', lease_token = NULL, lease_attempts = 0
                 FROM tasks
                 WHERE tasks.id = aggregation_jobs.task_id
                   AND tasks.task_id = $1
-                  AND aggregation_jobs.aggregation_job_id = $2",
+                  AND aggregation_jobs.aggregation_job_id = $2
+                  AND aggregation_jobs.lease_expiry = $3
+                  AND aggregation_jobs.lease_token = $4",
             )
             .await?;
         check_update(
             self.tx
                 .execute(
                     &stmt,
                     &[
-                        /* task_id */ &task_id.as_bytes(),
-                        /* aggregation_job_id */ &aggregation_job_id.as_bytes(),
+                        /* task_id */ &lease.leased().task_id.as_bytes(),
+                        /* aggregation_job_id */
+                        &lease.leased().aggregation_job_id.as_bytes(),
```

Comment:
```suggestion
                        /* aggregation_job_id */ &lease.leased().aggregation_job_id.as_bytes(),
```

Reply:
I definitely agree, but the auto-formatter insists on this layout :-/